File: solutions/three_sum/three_sum_bruteforce.py
def threesum_bruteforce(nums):
    result = []
    n = len(nums)
    for i in range(n):
        for j in range(i+1, n):
            for k in range(j+1, n):
                if nums[i] + nums[j] + nums[k] == 0:
                    triplet = sorted([nums[i], nums[j], nums[k]])
                    if triplet not in result:  # Avoid duplicates
                        result.append(triplet)
    return result

print(threesum_bruteforce([-1,0,1,2,-1,-4]))

File: solutions/three_sum/three_sum_bruteforce.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

vector<vector<int>> threeSumBruteforce(vector<int>& nums) {
    vector<vector<int>> result;
    int n = nums.size();

    for (int i = 0; i < n; i++) {
        for (int j = i+1; j < n; j++) {
            for (int k = j+1; k < n; k++) {
                if (nums[i] + nums[j] + nums[k] == 0) {
                    vector<int> triplet = {nums[i], nums[j], nums[k]};
                    sort(triplet.begin(), triplet.end());
                    if (find(result.begin(), result.end(), triplet) == result.end()) {
                        result.push_back(triplet); // Avoid duplicates
                    }
                }
            }
        }
    }
    return result;
}

int main() {
    vector<int> nums = {-1,0,1,2,-1,-4};
    vector<vector<int>> res = threeSumBruteforce(nums);

    for (auto &triplet : res) {
        for (int num : triplet) cout << num << " ";
        cout << endl;
    }
}
